### DISCUSSION TITLE
Verbesserte Kennzeichnung unverplanter Filme und Serien im Programmplaner

### DISCUSSION THREAD

**Discussion by jukey** [TVTower/TVTower#319] [Ideas]
Ich kaufe Filmlizenzen üblicherweise und verkaufe sie nach dem Senden normalerweise auch zügig wieder. Um, die Programmplanung zu vereinfachen und noch leichter zu sehen in welchen Kategorien noch nicht verplante Filme zu finden sind würde es mir sehr helfen, wenn schon die Kategorien/Ordner selbst farblich markiert wären.

1. Derzeit ist das Verhalten wie folgt:
![grafik](https://user-images.githubusercontent.com/295928/113043763-33ce8e80-919d-11eb-86dc-50504b5945e8.png) ![grafik](https://user-images.githubusercontent.com/295928/113043858-53fe4d80-919d-11eb-98fb-7f83dae024f2.png)
Direkt nach dem Erwerb neuer Lizenzen beim ersten laden des Programmplaners sind Kategorien/Ordner und die Filme selbst hellblau hinterlegt (In der Abbildung sind die Ordner (1) und Filme (2) nicht hellblau hinterlegt, da der Programmplaner nicht zum ersten Mal nach dem Besuch des Filmverleihs aufgesucht wurde, schätze ich?).
* Änderungsvorschlag:
Ich würde gern dauerhaft sehen, in welchen Kategorie-Ordnern noch Filme zu finden sind, die nicht verplant wurden. Können die Ordner und Filme entweder dauerhaft farblich hervorgehoben werden oder dies durch einen 'Filter' aktiviert- und deaktiviert werden?

2. Derzeit heben sich Serien im Programmplaner dadurch von Filmen ab, indem sie nachdem sie ausgewählt wurden, die einzelnen Folgen zur Auswahl anzeigen.
![grafik](https://user-images.githubusercontent.com/295928/113043944-6d9f9500-919d-11eb-9e8a-36748605a92d.png)
* Änderungsvorschlag: Serien sollten sich im Aussehen von Filmen unterscheiden, um klar zumachen, dass sie ausgewählt werden können und dann erst ihre Einzelfolgen anzeigen. Idealerweise sollten auch Serien eingefärbt werden, sofern sie noch ungeplante Folgen (3) enthalten.

Wie denkt ihr darüber?

**Comment by GWRon** [comment]
Serien(koepfe) sind lila...im Gegensatz zu Einzelprogrammen. 

Der aktuelle Quellcode hebt bereits such Serien blau hervor, sobald eine der Episoden noch "neu" ist (bspweise eine laufende Serie an Eigenproduktionen).

Ich glaube deine Punkte sind sozusagen seit gestern umgesetzt ;-)

Siehe dazu glaube (antworte gerade per Mail) issue #315

**Reply by jukey** [reply]
Die Umsetzung ging ja schnell! ;-)

Aber im Ernst: Danke für die Aufklärung bezüglich der Serienfarbe. Was mir nach wie vor passiert ist, dass die unverplanten Filme und Serien nicht hervorgehoben werden, wenn man nicht zuvor in der Filmagentur neue Lizenzen erworben hat. Ich würde die Kennzeichnung gern immer sehen oder eben per Filter aus- und einschalten.

So bleibt mir beim spielen nur, mich jeweils durch alle Ordner zu klicken, um Filme (weiß) und Serien mit unverplanten Folgen zu finden, die nicht verplant sind. Das mache ich so häufig, dass es mir den Aufwand der Eröffnung dieser Diskussion wert war. :)

**Comment by GWRon** [comment]
Wie gesagt...seit gestern wird eben hervorgehoben...auch bei Serien.

Es fehlt aber noch ein Commit, damit die Markierung nur noch nach Benutzung (das Aufheben eines neuen Programmblocks) verschwindet.

**Comment by nittka** [comment]
In der aktuell eingebauten Änderung bleiben Filme, Serien (und Einzelfolgen) so lange hellblau (d.h. neu hinzugekommen und noch nicht vom Spieler angefasst), bis sie (Filme und Einzelfolgen) mal vom Spieler angehoben wurden.
Für Filme und Einzelfolgen ist das gewünschte Verhalten also nun erfüllt (tatsächlich habe ich es noch nicht mit gekauften Serien ausprobiert). Wenn ich es richtig verstehe, passt die jetzige Umsetzung aber noch nicht ganz für Serien. Mit dem aktuellen Stand würde eine Serie nämlich als "geplant" markiert werden, wenn mindestens eine Folge geplant ist (und keine neue Folge existiert). Damit würde man an der Serie nicht erkennen, dass es noch unverplante Folgen gibt - und (wichtiger) sie ist auch nicht von Filmen zu unterscheiden.

Man könnte das nun umbauen und eine Serie erst dann als verplant markieren, wenn alle Folgen verplant sind. Ich denke, dass das tatsächlich einen Mehrwert hätte.
* Serie hellblau, wenn es noch neue Folgen gibt
* Serie lila und damit als Serie erkennbar, wenn  nicht alle Folgen verplant sind
* Serie verplant (und nicht von Film zu unterscheiden), wenn alle Folgen in der Programmplanung liegen - OK da ja alle Folgen schon im Programm sind

[Edit]: Ich habe gerade nochmal nachgeschaut. Die Kassettenfarbe unterscheidet sich bei Filmen und Serien auch bei vorhandener Neu- oder Geplantmarkierung (grau vs. lila). Die Unterscheidung ist also gewährleistet. Damit bleibt nur die Frage, welche Information bei Serien interessanter ist: es gibt ungeplante Folgen oder es gibt geplante Folgen. Persönlich bringt mir die erste Information wenig - eine Folge ist gerade zur besten Sendezeit gelaufen und die Aktualität is 0, bis zur vollen Aktualität dauert es viele Tage, also werde ich die nächste Ausstrahlung nicht jetzt planen wollen. Insofern bietet mir die Information, dass nicht alle Folgen verplant sind, keinen Mehrwert.

**Reply by jukey** [reply]
>  Insofern bietet mir die Information, dass nicht alle Folgen verplant sind, keinen Mehrwert.

Da stimme ich zu.

Ansonsten klingt es für mich, als wäre der initiale Vorschlag in der der neuesten, noch nicht veröffentlichten, Version enthalten. Wenn es eine gebaute Test-Version gibt, schaue ich es mir gern an.
